### DISCUSSION TITLE
How to keep the numbers in the header unchanged?

### DISCUSSION THREAD

**Discussion by OPEN1004MIND** [cpbotha/obsidian-ai-chat-as-md#3] [Ideas]
I confirmed that the plugin is working fine. thank you.
If I could make one change, it would be to keep the number of headers as they get lower and lower.
It would be great to have an optional setting to lower or keep the headers.

**Before Sample**

##### AI

***

**After Sample**
## AI

Thank you reading again.

**Comment by cpbotha** [comment]
Thank you very much for testing.

Please study https://github.com/cpbotha/obsidian-ai-chat-as-md/blob/master/docs/example_nesting.md for an explanation why I nest the headings like this.

The nesting is a core part of what makes this plugin interesting. You can go back up the thread, and then branch out into a different conversation but keep all of your other branches.

At the moment it's not something I plan to implement, but I will think a bit if there's an elegant way for me to support non-nesting headings, like you show in your example, without affecting the main flow.

**Comment by texastoland** [comment]
I wanted to FR the same. It's limited to 5 messages per branch with mental overhead to remember nesting level:

```md
# First message: joke request

## AI: first reply with joke (not so funny...)

### More serious

#### AI: more serious

### More funny

#### AI: attempt 2 at funnier (simply retry)

#### AI: attempt 1 at funnier
```

Maybe https://github.com/markdown-it/markdown-it-container could wrap branches:

```md
# First message: joke request

# AI: first reply with joke (not so funny...)

::: ai-branch

# More serious

# AI: more serious

:::
::: ai-branch

# More funny

:::: ai-branch

# AI: attempt 2 at funnier (simply retry)

::::
:::: ai-branch

# AI: attempt 1 at funnier

::::
:::
```

I also considered lists or blockquotes. None are ideal. Just brainstorming 💭
